### PR TITLE
[Editor] Add the delete icon for each Siddhi file which is under the work space explorer.

### PR DESCRIPTION
**Pull request by SinthujanSintha**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js [hunk 1]
```diff
@@ -38,21 +38,27 @@ define(['require', 'lodash', 'jquery', 'log', 'backbone', 'file_browser',
...
                     var notification_container = this.notification_container;
-                    var workspaceServiceURL = app.config.services.workspace.endpoint;
-                    var activeTab = app.tabController.activeTab;
-                    var siddhiFileEditor = activeTab.getSiddhiFileEditor();
-                    var content = siddhiFileEditor.getContent();
-                    var providedFileName = activeTab.getTitle();
-                    var trimmedSiddhiAppName = providedFileName;
-                    if (checkEndsWithSiddhi(trimmedSiddhiAppName)) {
-                        trimmedSiddhiAppName = trimmedSiddhiAppName.slice(0, -7);
+                    // var workspaceServiceURL = app.config.services.workspace.endpoint;
```
Comment (BuddhiWathsala) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js]: Please remove this commented line.

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js [hunk 2]
```diff
@@ -38,21 +38,27 @@ define(['require', 'lodash', 'jquery', 'log', 'backbone', 'file_browser',
...
+                        trimmedSiddhiAppName = trimSiddhiAppName(providedFileName);
+                        fileTab = app.tabController.getTabFromTitle(trimmedSiddhiAppName);
+                    } else {
+                        fileTab = app.tabController.activeTab;
+                        providedFileName = fileTab.getTitle();
+                        trimmedSiddhiAppName = trimSiddhiAppName(providedFileName);
                     }
-
+                    // var siddhiFileEditor = fileTab.getSiddhiFileEditor();
+                    // var content = siddhiFileEditor.getContent();
```
Comment (BuddhiWathsala) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js]: Please remove this commented lines.

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/file-browser/file-browser.js [hunk 3]
```diff
@@ -36,6 +36,43 @@ define(['jquery', 'backbone', 'lodash', 'log', /** void module - jquery plugin *
...
             this._showBundles = _.get(config, 'showBundles');
             this._multiSelect = _.get(config, 'multiSelect', false);
+            this._deleteIcon = _.get(config, 'deleteIcon', false);
+            var self = this;
+
+            /**
+             * custom plugin for jstree to add delete icon for each file in workspace-explorer.
+             */
+            (function ($, undefined) {
+                "use strict";
```
Comment (BuddhiWathsala) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/file-browser/file-browser.js]: What is the usage of this?

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js [hunk 4]
```diff
@@ -38,21 +38,27 @@ define(['require', 'lodash', 'jquery', 'log', 'backbone', 'file_browser',
...
+                    // var workspaceServiceURL = app.config.services.workspace.endpoint;
+                    var providedFileName;
+                    var trimmedSiddhiAppName;
+                    //check fileName is available or not.
+                    if (!_.isNil(fileName)) {
+                        providedFileName = fileName;
+                        trimmedSiddhiAppName = trimSiddhiAppName(providedFileName);
+                        fileTab = app.tabController.getTabFromTitle(trimmedSiddhiAppName);
+                    } else {
+                        fileTab = app.tabController.activeTab;
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js]: Under what circumstances filename becomes null? solve the issue. Here, you have captured the active tab and showing the name which is not the correct way to handle this

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/workspace/workspace.js [hunk 5]
```diff
@@ -550,7 +550,7 @@ define(['ace/ace', 'jquery', 'lodash', 'log','dialogs','./service-client','welco
                 if(_.isNil(this._deleteFileDialog)){
                     this._deleteFileDialog = new Dialogs.DeleteConfirmDialog(app);
                 }
-                this._deleteFileDialog.render();
+                this._deleteFileDialog.render(options);
```
Comment (niveathika) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/workspace/workspace.js]: Options should be passed when creating the dialog not during render

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js (hunk 1 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js]: 80a9a2d1fcd5fc6e4c241a17de5a4016e0107a2b this commit fix the above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js (hunk 2 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js]: 80a9a2d this commit fix the above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/file-browser/file-browser.js (hunk 3 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/file-browser/file-browser.js]: 80a9a2d this commit fix the above request change

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/workspace/workspace.js (hunk 5 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/workspace/workspace.js]: This Dialog object is only created once So filename is parsed through render

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js (hunk 4 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js]: Explained in 4bdd75d

Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/delete-confirm-dialog.js]: Ack

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/workspace/workspace.js (hunk 5 above)
Comment (niveathika) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/workspace/workspace.js]: Ack

**Comment by niveathika** [comment]
@BuddhiWathsala Seems @SinthujanSintha has addressed your comments, please folllowup

**Pull request merged**